static/src/javascripts/projects/commercial/commercial-metrics.ts

```diff
@@ -0,0 +1,87 @@
+import { EventTimer } from '@guardian/commercial-core';
+import { log } from '@guardian/libs';
+import config_ from '../../lib/config';
+
+// This is really a hacky workaround ⚠️
+const config = config_ as {
+	get: (s: string, d?: unknown) => unknown;
+};
+
+type CommercialMetrics = {
+	browser_id?: string;
+	page_view_id: string;
+	received_timestamp: string;
+	received_date: string;
+	platform: string;
+	metrics: Metrics[];
+	properties: Properties[];
+};
+
+type Metrics = {
+	name: string;
+	value: number;
+};
+
+type Properties = {
+	name: string;
+	value: string;
+};
+
+let logged = false;
+
+const isDev = Boolean(config.get('page.isDev', false));
+const devProperties: Properties[] = isDev
+	? [{ name: 'isDev', value: window.location.hostname }]
+	: [];
+
+const endpoint = isDev
+	? '//performance-events.code.dev-guardianapis.com/commercial-metrics'
+	: '//performance-events.guardianapis.com/commercial-metrics';
+
+const logData = (): void => {
+	if (logged) return;
+	if (!window.guardian.ophan) return;
+	if (document.visibilityState !== 'hidden') return;
+
+	const timestamp = new Date().toISOString();
+	const date = timestamp.slice(0, 10);
+	const eventTimer = EventTimer.get();
+	const events = eventTimer.events;
+
+	const properties: Properties[] = Object.entries(eventTimer.properties)
+		.map((property) => {
+			const [name, value] = property;
+			return { name, value: String(value) };
+		})
+		.concat(devProperties);
+
+	const metrics: Metrics[] = events.map((event) => {
+		return { name: event.name, value: Math.ceil(event.ts) };
+	});
+
+	const browser_id = config.get('ophan.browserId') as string | undefined;
+
+	const commercialMetrics: CommercialMetrics = {
+		browser_id,
+		page_view_id: window.guardian.ophan.pageViewId,
+		received_timestamp: timestamp,
+		received_date: date,
```

Comment:
These should really come from the server-side, but it seems like we’d need to look more deeply into the back-end config.

Reply:
As long as we're considering moving `received_timestamp` and `received_date` server side, is there a reason to store both values? Or is it worth considering dropping `received_date` altogether since it can be derived from `received_timestamp`?

Reply:
As mentioned [in the data PR][1]:

> The reason for having both is that the buckets are partitioned by `received_date`, meaning you’d only read the data for a singular day by default. We should be moving this logic to Fastly, which would ensure that the two dates come from the same value, but there is value in having both.

[1]: https://github.com/guardian/data-platform-models/pull/1235#discussion_r638798734